### ISSUE TITLE
Labels stacked on each others

### ISSUE THREAD

**Issue by BrunoTetra** [chartjs/chartjs-plugin-datalabels#78]
Hi everyone, 
i have a "problem" using this plugin : here is an example
![image](https://user-images.githubusercontent.com/37797493/43722362-602a01f6-9995-11e8-8292-d37fde8ee974.png)

As you can see, some labels are stacked on each others and are unreadable. (28/30 - 15/16)

Is there any way to prevent this natively ?

My options : 
```
{
  responsive: true,
  maintainAspectRatio: false,
  spanGaps: true,
  title: {
    display: true,
    fontSize: 15,
    fontFamily: "Tahoma",
    fontStyle: "Bold",
    text: "title"
  },
  animation: {},
  layout: {
    padding: {left: 0, right: 0, top: 0, bottom: 30}
  },
  legend: {
    position: "bottom"
  },
  scales: {
    xAxes: [{ticks: {autoSkip: false}}],
    yAxes: [{ticks: {beginAtZero: true, dataTag: true}}]
  },
  plugins: {
    datalabels: {display: true, align: "end", anchor: "end"}
  }
}
```
Here is a little jsfiddle version of the example above : https://jsfiddle.net/8hp5fdbe/30/

Thank's in advance
- Bruno

**Comment by simonbrunel** [comment]
@BrunoTetra / @grayniall sorry for the late reply. I just commented a similar issue (#59) with a (limited) workaround (see this [fiddle](https://jsfiddle.net/8hp5fdbe/37/)). I'm still trying to figure out how to implement a decent positioning solution to prevent labels to overlap.

@BrunoTetra in your case, what would you expect when 2 labels overlap?

**Comment by BrunoTetra** [comment]
Hi @simonbrunel ,

i'm not on the project anymore but i think my former collegue has found a temporary solution.
When you put another point on the chart: if the label is too close to enother one, don't show it.
I don't have access to the code right now, but if i figure it out, i'll share it with you

**Comment by simonbrunel** [comment]
Hiding overlapping labels is now implemented (92a926fc32bd) and will be released in [v0.5.0](/chartjs/chartjs-plugin-datalabels/milestone/5).

Attaching the associated build if some of you want to test and confirm that it works as expected :)

[chartjs-plugin-datalabels-92a926f.zip](https://github.com/chartjs/chartjs-plugin-datalabels/files/2612397/chartjs-plugin-datalabels-92a926f.zip)

**Comment by simonbrunel** [comment]
Released in [v0.5.0](/chartjs/chartjs-plugin-datalabels/releases/tag/v0.5.0) ([fiddle](https://jsfiddle.net/hso3fk8j/))
